### PR TITLE
Added Pagination to dynamodb queries

### PR DESCRIPTION
**Pull request by Jlrine2**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.8.2]
+## Changed
+- Refactored dynamodb interactions
+  - `dynomo.py` in the api code now manages all dynamodb interactions for the api
```
Comment (asjohnston-asf) [inline CHANGELOG.md:10]: `dynamo.py` typo

File: apps/api/src/hyp3_api/dynamo.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    )
+    jobs = response['Items']
+
+    while 'LastEvaluatedKey' in response:
+        response = table.query(
+            IndexName='user_id',
+            KeyConditionExpression=key_expression,
+            FilterExpression=filter_expression,
+            ExclusiveStartKey=response['LastEvaluatedKey'],
+            )
```
Comment (asjohnston-asf) [inline apps/api/src/hyp3_api/dynamo.py:57]: outdent one level

File: apps/start-execution/src/start_execution.py [hunk 3]
```diff
@@ -24,7 +24,14 @@ def get_pending_jobs():
         IndexName='status_code',
         KeyConditionExpression=Key('status_code').eq('PENDING'),
     )
-    return response['Items']
+    pending_jobs = response['Items']
+    while 'LastEvaluatedKey' in response:
+        response = table.query(
+            IndexName='status_code',
+            KeyConditionExpression=Key('status_code').eq('PENDING'),
+            )
```
Comment (asjohnston-asf) [inline apps/start-execution/src/start_execution.py:32]: outdent one level

File: tests/api/test_dynamo.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+            'request_time': '2000-01-01T00:00:00+00:00'
+        },
+    ]
+    for item in table_items:
+        tables['jobs_table'].put_item(Item=item)
+
+    assert dynamo.get_job('job1') == table_items[0]
+    assert dynamo.get_job('job2') == table_items[1]
+    assert dynamo.get_job('job3') == table_items[2]
+
```
Comment (asjohnston-asf) [inline tests/api/test_dynamo.py:208]: `assert dynamo.get_job('foo') == None`

File: tests/api/test_dynamo.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        {
+            'user_id': 'user2',
+            'max_jobs_per_user': 15
+        },
+    ]
+    for item in table_items:
+        tables['users_table'].put_item(Item=item)
+
+    assert dynamo.get_user('user1') == table_items[0]
+    assert dynamo.get_user('user2') == table_items[1]
```
Comment (asjohnston-asf) [inline tests/api/test_dynamo.py:225]: `assert dynamo.get_user('foo') == None`

File: tests/api/test_dynamo.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            'request_time': '2000-01-01T00:00:00+00:00'
+        },
+        {
+            'job_id': 'job3',
+            'user_id': 'user2',
+            'job_type': 'fake job type',
+            'status_code': 'fake job stats',
+            'request_time': '2000-01-01T00:00:00+00:00'
+        },
+    ]
```
Comment (asjohnston-asf) [inline tests/api/test_dynamo.py:27]: Looks like we can define one set of mock records (global or helper function) that can support all of the tests and eliminate having this declaration 6 times

Comment (Jlrine2) [reply tests/api/test_dynamo.py:27]: probably, but I like that the tests assetions are close to the data declaration, I don't have to scroll to see what I'm putting in the table

File: tests/api/test_dynamo.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+    for item in table_items:
+        tables['jobs_table'].put_item(Item=item)
+
+    response = dynamo.query_jobs('user1')
+    assert len(response) == 2
+    assert [i for i in response if i not in table_items[:2]] == []
```
Comment (asjohnston-asf) [inline tests/api/test_dynamo.py:33]: is this just `assert response == table_items[0:2]`, e.g. we got the first and second items back in our list?

Comment (asjohnston-asf) [reply tests/api/test_dynamo.py:33]: or even `assert dynamo.query_jobs('user1') == table_items[0:2]`

File: apps/api/src/hyp3_api/handlers.py [hunk 8]
```diff
@@ -67,65 +65,41 @@ def post_jobs(body, user):
...
     for job in body['jobs']:
         job['job_id'] = str(uuid4())
         job['user_id'] = user
         job['status_code'] = 'PENDING'
         job['request_time'] = request_time
-        if not body.get('validate_only'):
-            job = convert_floats_to_decimals(job)
-            table.put_item(Item=job)
-
+        jobs.append(convert_floats_to_decimals(job))
```
Comment (asjohnston-asf) [inline apps/api/src/hyp3_api/handlers.py:74]: How about moving `convert_floats_to_decimals` inside of `dynamo.put_jobs()`?

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**